LINK VISUALISASI TABLEAU : https://public.tableau.com/views/INSYAALLAHFIX/Dashboard1?:language=en-US&publish=yes&:sid=&:display_count=n&:origin=viz_share_link

# SMART dan PROBLEM STATEMENT

**Case** = Production House A merupakan start up produsen film yang sedang berkembang, dalam proses perintisannya membangun perusahaan mereka ingin membangun portofolio film dengan rating yang lebih tinggi dari rata-rata rating film Indonesia untuk membangun kepercayaan calon investor. Production House A berencana ingin mengetahui karakteristik film Indonesia berdasarkan genre film dan durasi untuk nantinya menjadi bahan acuan mereka untuk membuat film.


1. **Specific**: Mengetahui Karakteristik film Indonesia yang memiliki rating diatas rata-rata.

<!-- 2. **Measurable**: Menaikkan rating film mereka >= rata-rata rating film indonesia -->

3. **Achievable**: Dengan mengetahui rata-rata rating film Indonesia, mengetahui karakteristik film Indonesia yang memiliki rating diatas rata-rata.

4. **Relevant**: Mengetahui karakteristik film yang memiliki rating diatas rata-rata dapat menjadi acuan Production House A untuk membuat film

5. **Time-Bound**: Q3 pada 2024

**Statement** : Mengetahui karakteristik film dengan rating diatas rata-rata dapat menjadi acuan bagi Production House A membuat film yang memiliki rating tinggi sebagai portofolio mereka untuk membangun kepercayaan calon investor.



**Problem Statement**: `Mengetahui Karakteristik film Indonesia yang memiliki rating diatas rata-rata`

  - Penjabaran masalah dengan metode 5W+1H:

 1. Berapa rata-rata rating film Indonesia ? (menggunakan deskriptif)
 2. Apa genre film yang memiliki rating diatas rata-rata ?(Menggunakan visualisasi )
 3. siapa director film dengan rating diatas rata-rata ? (Menggunakan visualisasi)
 4. Berapa range durasi film yang memiliki rating diatas rata-rata? (Menggunakan visualisasi)
 5. Pada tahun berapa yang memiliki rating diatas rata-rata?(Menggunakan visualisasi)
 6. Apakah genre film memiliki pengaruh terhadap rating? (Menggunakan inferensial)
 7. Apakah lama durasi film memiliki pengaruh terhadap rating ? (Menggunakan Inferensial)

# DATA LOADING

In [63]:
import pandas as pd

# lokasi file 
file_path = "/Users/daniyalkahfi/Desktop/Data/Milestone 1/indonesian_movies.csv"

# Baca file CSV menjadi DataFrame
df = pd.read_csv(file_path)

# Tampilkan beberapa baris pertama dari DataFrame
df.head()


title  year  \
0       #FriendButMarried 2  2020   
1                  4 Mantan  2020   
2  Aku Tahu Kapan Kamu Mati  2020   
3               Anak Garuda  2020   
4                 Dignitate  2020   

                                         description      genre rating  \
0  Ayudia (Mawar De Jongh) is not satisfied enoug...  Biography    13+   
1  Sara, Airin, Rachel, and Amara were accidental...   Thriller    17+   
2  After apparent death, Siena is able to see sig...     Horror    13+   
3  Good Morning Indonesia, a school for poor orph...  Adventure    13+   
4  Alfi (Al Ghazali) meets Alana (Caitlin Halderm...      Drama    17+   

   users_rating votes   languages          directors  \
0           6.5   120  Indonesian      Rako Prijanto   
1           6.4     8  Indonesian      Hanny Saputra   
2           5.4    17  Indonesian  Hadrah Daeng Ratu   
3           9.1    27  Indonesian       Faozan Rizal   
4           7.6    33  Indonesian       Fajar Nugros   

                                              actors  runtime  
0  ['Adipati Dolken', 'Mawar Eva de Jongh', 'Vonn...  100 min  
1  ['Ranty Maria', 'Jeff Smith', 'Melanie Berentz...   80 min  
2  ['Natasha Wilona', 'Ria Ricis', 'Al Ghazali', ...   92 min  
3  ['Tissa Biani Azzahra', 'Violla Georgie', 'Aji...  129 min  
4  ['Al Ghazali', 'Caitlin Halderman', 'Giorgino ...  109 min

In [64]:
#Mendapatkan Informasi dari dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1272 entries, 0 to 1271
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   title         1272 non-null   object 
 1   year          1272 non-null   int64  
 2   description   840 non-null    object 
 3   genre         1236 non-null   object 
 4   rating        376 non-null    object 
 5   users_rating  1272 non-null   float64
 6   votes         1272 non-null   object 
 7   languages     1272 non-null   object 
 8   directors     1265 non-null   object 
 9   actors        1272 non-null   object 
 10  runtime       869 non-null    object 
dtypes: float64(1), int64(1), object(9)
memory usage: 109.4+ KB


Dari informasi dataset diketahui bahwa ada 1272 data dengan 11 kolom. Terdapat juga missing value pada beberapa kolom yang perlu dihapus (kecuali description) dan perubahan tipe data yang sesuai dengan peruntukannya

# DATA CLEANING

### Merubah Nama Kolom

In [65]:
# merubah nama kolom agar lebih mudah dipahami
df = df.rename(columns = {'runtime' : 'duration'})
df.head()

title  year  \
0       #FriendButMarried 2  2020   
1                  4 Mantan  2020   
2  Aku Tahu Kapan Kamu Mati  2020   
3               Anak Garuda  2020   
4                 Dignitate  2020   

                                         description      genre rating  \
0  Ayudia (Mawar De Jongh) is not satisfied enoug...  Biography    13+   
1  Sara, Airin, Rachel, and Amara were accidental...   Thriller    17+   
2  After apparent death, Siena is able to see sig...     Horror    13+   
3  Good Morning Indonesia, a school for poor orph...  Adventure    13+   
4  Alfi (Al Ghazali) meets Alana (Caitlin Halderm...      Drama    17+   

   users_rating votes   languages          directors  \
0           6.5   120  Indonesian      Rako Prijanto   
1           6.4     8  Indonesian      Hanny Saputra   
2           5.4    17  Indonesian  Hadrah Daeng Ratu   
3           9.1    27  Indonesian       Faozan Rizal   
4           7.6    33  Indonesian       Fajar Nugros   

                                              actors duration  
0  ['Adipati Dolken', 'Mawar Eva de Jongh', 'Vonn...  100 min  
1  ['Ranty Maria', 'Jeff Smith', 'Melanie Berentz...   80 min  
2  ['Natasha Wilona', 'Ria Ricis', 'Al Ghazali', ...   92 min  
3  ['Tissa Biani Azzahra', 'Violla Georgie', 'Aji...  129 min  
4  ['Al Ghazali', 'Caitlin Halderman', 'Giorgino ...  109 min

### Handling Missing Value

In [66]:
# Mengecek missing value dari dataframe
df.isnull().sum()

title             0
year              0
description     432
genre            36
rating          896
users_rating      0
votes             0
languages         0
directors         7
actors            0
duration        403
dtype: int64

In [67]:
# menghapus missing value pada kolom genre dan duration
df = df.dropna(subset=['genre', 'duration', 'directors', 'rating'])
df.isnull().sum()

title            0
year             0
description     14
genre            0
rating           0
users_rating     0
votes            0
languages        0
directors        0
actors           0
duration         0
dtype: int64

tidak dilakukan handling missing value pada kolom description karena kolom tersebut tidak berpengaruh terhadap pengolahan data statistik. Hanya dijadikan bahan pendukung dalam memahami dataset

### Penyesuaian Data Types

In [68]:
#mengecek tipe data setiap kolom
df.dtypes

title            object
year              int64
description      object
genre            object
rating           object
users_rating    float64
votes            object
languages        object
directors        object
actors           object
duration         object
dtype: object

### Drop Kolom yang Tidak Relevan

In [69]:
#menghapus kolom rating dan votes
df.drop(columns=['rating', 'votes'], inplace=True)

### Penghapusan Karakter Serta Merubah Tipe Data

In [70]:
#Merubah tipe data duration menjadi integer
df['duration'] = df['duration'].str.replace(' min', '').astype(int)
#Merubah tipe data year menjadi datetime
df['year'] = pd.to_datetime(df['year'], format='%Y')


#merubah nama kolom duration agar diketahui dalam satuan minutes
df.rename(columns={'users_rating' : 'rating','duration': 'duration_minutes'}, inplace=True)
df.dtypes


title                       object
year                datetime64[ns]
description                 object
genre                       object
rating                     float64
languages                   object
directors                   object
actors                      object
duration_minutes             int64
dtype: object

### Duplicates Value Checking

In [71]:
# mengecek data duplikat
df.duplicated().sum()

0

tidak adanya data duplikat pada dataframe

### Filter Data Berdardasarkan Tahun >= 2005

In [72]:
# Dilakukan filter data year mulai dari tahun 2005, karena dataset memuat film dari tahun 1980. 
# Data menjadi tidak valid jika range tahun terlalu jauh, karena teknologi perfilman sebelum tahun 2000 masih terlalu kuno dibanding sekarang

df = df[df['year'].dt.year >= 2005]
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 289 entries, 0 to 969
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   title             289 non-null    object        
 1   year              289 non-null    datetime64[ns]
 2   description       276 non-null    object        
 3   genre             289 non-null    object        
 4   rating            289 non-null    float64       
 5   languages         289 non-null    object        
 6   directors         289 non-null    object        
 7   actors            289 non-null    object        
 8   duration_minutes  289 non-null    int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(6)
memory usage: 22.6+ KB


Dilakukan filter data agar data yang didapatkan lebih valid karena dalam dataset tahun rilis film sejak 1980 - 2020. Rentang tahun yang terlalu memiliki gap teknologi film yang terlalu usang. Selain itu, preferensi penonton juga sudah jauh berubah dibandingkan film pada tahun 1980

### Menambahkan Kolom Duration Category

In [73]:
# Perencanaan Pembagian range waktu
dur_max = df['duration_minutes'].max()
dur_min = df['duration_minutes'].min()
dur_diff = df['duration_minutes'].max() - df['duration_minutes'].min()
print (dur_max)
print (dur_min)
print (dur_diff/4)

181
71
27.5


In [74]:
# Klasifikasi duration_minutes ke dalam 4 kategori dengan range 70 sampai 200 dengan kelipatan 30 menit
df['duration_category'] = pd.cut(df['duration_minutes'], bins=[70, 100, 130, 160, 200], labels=['70-100', '101-130', '131-160', '161-190'])

df

title       year  \
0         #FriendButMarried 2 2020-01-01   
1                    4 Mantan 2020-01-01   
2    Aku Tahu Kapan Kamu Mati 2020-01-01   
3                 Anak Garuda 2020-01-01   
4                   Dignitate 2020-01-01   
..                        ...        ...   
916        Malam Jumat Kliwon 2007-01-01   
919   Miracle: Menantang Maut 2007-01-01   
933                    9 Naga 2006-01-01   
953                  Pocong 2 2006-01-01   
969                    Mirror 2005-01-01   

                                           description      genre  rating  \
0    Ayudia (Mawar De Jongh) is not satisfied enoug...  Biography     6.5   
1    Sara, Airin, Rachel, and Amara were accidental...   Thriller     6.4   
2    After apparent death, Siena is able to see sig...     Horror     5.4   
3    Good Morning Indonesia, a school for poor orph...  Adventure     9.1   
4    Alfi (Al Ghazali) meets Alana (Caitlin Halderm...      Drama     7.6   
..                                                 ...        ...     ...   
916                                                NaN     Horror     2.2   
919  Kinar got alphabetic code KITA MATI that indic...     Horror     3.6   
933  Three friends since childhood earn a living by...      Crime     6.5   
953  Two sisters find themselves haunted by an evil...     Horror     6.5   
969  A high school girl named Kikan began to see de...     Horror     6.9   

      languages          directors  \
0    Indonesian      Rako Prijanto   
1    Indonesian      Hanny Saputra   
2    Indonesian  Hadrah Daeng Ratu   
3    Indonesian       Faozan Rizal   
4    Indonesian       Fajar Nugros   
..          ...                ...   
916  Indonesian   Nayato Fio Nuala   
919  Indonesian  Helfi C.H. Kardit   
933  Indonesian     Rudy Soedjarwo   
953  Indonesian     Rudy Soedjarwo   
969  Indonesian      Hanny Saputra   

                                                actors  duration_minutes  \
0    ['Adipati Dolken', 'Mawar Eva de Jongh', 'Vonn...               100   
1    ['Ranty Maria', 'Jeff Smith', 'Melanie Berentz...                80   
2    ['Natasha Wilona', 'Ria Ricis', 'Al Ghazali', ...                92   
3    ['Tissa Biani Azzahra', 'Violla Georgie', 'Aji...               129   
4    ['Al Ghazali', 'Caitlin Halderman', 'Giorgino ...               109   
..                                                 ...               ...   
916  ['Daffy Ariaga', 'Abimana Aryasatya', 'Nadiah ...                90   
919  ['Abimana Aryasatya', 'Ivy Batuta', 'Nadila Er...                98   
933  ['Donny Alamsyah', 'Fauzi Baadila', 'Dorman Bo...               104   
953  ['Revalina S. Temat', 'Ringgo Agus Rahman', 'H...                98   
969  ['Nirina Zubir', 'Jonathan Lesmana', 'Ichi Nur...               112   

    duration_category  
0              70-100  
1              70-100  
2              70-100  
3             101-130  
4             101-130  
..                ...  
916            70-100  
919            70-100  
933           101-130  
953            70-100  
969           101-130  

[289 rows x 10 columns]

# ANALISIS DAN PERHITUNGAN

In [75]:
from scipy.stats import f_oneway
from scipy.stats import spearmanr

### Berapa rata-rata rating film Indonesia ? (menggunakan deskriptif)

In [76]:
mean_rating= df['rating'].mean().round(2)
median_rating= df['rating'].median().round(2)
modus_rating= df['rating'].mode()[0]
std_rating = df['rating'].std().round(2)

print('Mean dari rating film Indonesia adalah :', mean_rating)
print('Median dari rating film Indonesia adalah :', median_rating)
print('Modus dari rating film Indonesia adalah :', modus_rating)
print('Standar Deviasi dari rating adalah :', std_rating)

Mean dari rating film Indonesia adalah : 6.35
Median dari rating film Indonesia adalah : 6.6
Modus dari rating film Indonesia adalah : 7.0
Standar Deviasi dari rating adalah : 1.43


Data rating terindikasi tidak terdistribusi normal, karena mean dan mediannya tidak sama.

In [77]:
# Mencari nilai rating tertinggi
max_rating = df['rating'].max()
print ('rating tertinggi film Indonesia adalah = ', max_rating)

rating tertinggi film Indonesia adalah =  9.4


### Apakah genre film memiliki pengaruh terhadap rating? (Menggunakan inferensial)

Mengetahui tipe distribusi rating

In [78]:
print("Skewness pada rating :", df['rating'].skew())
print("Kurtosis pada rating :", df['rating'].kurtosis())

Skewness pada rating : -1.0035265238346442
Kurtosis pada rating : 1.1553072654601517


Data rating tidak terdistribusi normal karena memiliki skewness negatif dan kurtosis != 0

In [79]:
# Menghitung korelasi Pearson antara genre dan rating
# menggunakan pearson karena data tidak terdistribusi normal
rho, p_value = spearmanr(df['genre'], df['rating'])

print("Korelasi spearmanr:", rho)
print("Nilai p-value:", p_value)

if p_value < 0.05:
    print("Terdapat korelasi yang signifikan antara genre dan rating.")
else:
    print("Tidak terdapat korelasi yang signifikan antara genre dan rating.")



Korelasi spearmanr: -0.2028340493815295
Nilai p-value: 0.0005216862035069283
Terdapat korelasi yang signifikan antara genre dan rating.


- H0 = Genre film TIDAK BERPENGARUH terhadap Rating
- H1 = Genre film BERPENGARUH terhadap Rating
- Dari hasil nilai korelasi spearmanr <= 0 dan p-value < 0,05 maka, H1 diterima (Genre film mempengaruhi rating)

## Apakah durasi film memiliki pengaruh terhadap rating ? (Menggunakan Inferensial)

In [80]:
# Menghitung korelasi Pearson antara duration_minutes dan rating
# menggunakan pearson karena data tidak terdistribusi normal
rho, p_value = spearmanr(df['duration_minutes'], df['rating'])

print("Korelasi spearmanr:", rho)
print("Nilai p-value:", p_value)

if p_value < 0.05:
    print("Terdapat korelasi yang signifikan antara duration_minutes dan rating.")
else:
    print("Tidak terdapat korelasi yang signifikan antara duration_minutes dan rating.")



Korelasi spearmanr: 0.330108098068919
Nilai p-value: 8.951449848385369e-09
Terdapat korelasi yang signifikan antara duration_minutes dan rating.


- H0 = Durasi film TIDAK BERPENGARUH terhadap rating
- H1 = Durasi film BERPENGARUH terhadap rating
- Dari hasil nilai korelasi spearmanr <= 0 dan p-value < 0,05 maka, H1 diterima (Durasi Film MEMPENGARUHI Rating)

## MENJADIKAN DATAFRAME SEBAGAI FILE CSV

In [81]:
df.to_csv('Data_Cleaned_Film_edit.csv')

In [82]:
df.head()

title       year  \
0       #FriendButMarried 2 2020-01-01   
1                  4 Mantan 2020-01-01   
2  Aku Tahu Kapan Kamu Mati 2020-01-01   
3               Anak Garuda 2020-01-01   
4                 Dignitate 2020-01-01   

                                         description      genre  rating  \
0  Ayudia (Mawar De Jongh) is not satisfied enoug...  Biography     6.5   
1  Sara, Airin, Rachel, and Amara were accidental...   Thriller     6.4   
2  After apparent death, Siena is able to see sig...     Horror     5.4   
3  Good Morning Indonesia, a school for poor orph...  Adventure     9.1   
4  Alfi (Al Ghazali) meets Alana (Caitlin Halderm...      Drama     7.6   

    languages          directors  \
0  Indonesian      Rako Prijanto   
1  Indonesian      Hanny Saputra   
2  Indonesian  Hadrah Daeng Ratu   
3  Indonesian       Faozan Rizal   
4  Indonesian       Fajar Nugros   

                                              actors  duration_minutes  \
0  ['Adipati Dolken', 'Mawar Eva de Jongh', 'Vonn...               100   
1  ['Ranty Maria', 'Jeff Smith', 'Melanie Berentz...                80   
2  ['Natasha Wilona', 'Ria Ricis', 'Al Ghazali', ...                92   
3  ['Tissa Biani Azzahra', 'Violla Georgie', 'Aji...               129   
4  ['Al Ghazali', 'Caitlin Halderman', 'Giorgino ...               109   

  duration_category  
0            70-100  
1            70-100  
2            70-100  
3           101-130  
4           101-130

# KESIMPULAN

1. Rating film Indonesia dari tahun 2005 - 2020 memiliki rata-rata 6.35. Untuk itu, Production House A harus membuat film dengan target rating > 6.35.
2. Film dengan Genre Romance, History, Biography, Drama, Crime, dan Comedy memiliki rating diata rata-rata film Indonesia. Romance menjadi film dengan genre tertinggi dengan nilai 7.0. Production House A dapat memilih salah satu dari genre tersebut dalam produksi film selanjutnya.
3. Directors film dengan rata-rata rating tertinggi adalah Ihdar Nur dengan nilai 8.8. Production House A dapat menjadikan karya-karya Ihdar Nur sebagai acuan dalam pembuatan film, seperti pola cara penulisan cerita, penyampaian cerita, teknik pengambilan video, dan cara memasarkan karya-karyanya.
4. Film dengan rentang durasi 161-190 menit menjadi film dengan rata-rata rating terbaik dengan nilai 7,1. Dalam rentang durasi tersebut sebuah film dapat disampaikan ceritanya dengan baik, alur cerita tidak terlalu cepat/lambat dan penonton tidak lelah untuk menikmati serta memahami makna film sampai akhir.
4. Karya-karya film pada tahun 2016 menjadi rata-rata rating terbaik dengan nilai 7.0. Film yang rilis pada tahun tersebut banyak yang disukai penikmat film Indonesia. Production House A dapat menjadikan referensi film-film di tahun 2016 sebagai acuan dalam pembuatan film.
5. Berdasarkan perhitungan statistik inferensial dengan metode Spearman, didapatkan hasil Korelasi spearmanr: -0.2028340493815295 dan Nilai p-value: 0.0005216862035069283. Artinya, pemilihan genre film berpengaruh signifikan terhadap rating yang akan didapatkan. Berdasarkan informasi sebelumnya, penikmat film Indonesia paling menyukai film dengan genre Romance, jadi Production House A disarankan membuat film dengan genre Romance agar peluang mendapatkan rating diatas rata-rata menjadi lebih besar.
6. Dilakukan pula pengujian korelasi antara rating dengan duration film. Hasil uji Korelasi spearmanr: 0.330108098068919 dan Nilai p-value: 8.951449848385369e-09. Artinya, penentuan duration film berkorelasi terhadap rating. Dari data sebelumnya, penikmat film Indonesia menyukai film dengan rentang waktu 161-190 menit. Dengan rentang waktu tersebut, film dapat menyampaikan cerita dengan detail, runut, dan tidak membuat penonton lelah/bosan karena terlalu menonton. 


### STATEMENT DAN INSIGHT

Dalam usaha membuat film dengan rating diatas rata-rata Film Indonesia (6,35)Production House A disarankan membuat film dengan genre Romance dengan durasi 161-190 menit, serta dalam proses pembuatannya dapat menjadikan karya-karya milik director Ihdar Nur dan film-film yang rilis pada tahun 2016 sebagai acuan pola penulisan cerita, penyampaian cerita, teknik pengambilan video, dan cara memasarkan karya-karyanya.